classes/mutex/PgAdvisoryMutex.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace malkusch\lock\mutex;
+
+use malkusch\lock\exception\LockAcquireException;
+use malkusch\lock\exception\LockReleaseException;
+
+/**
+ * @author Dmitrii Korotovskii <dmitry@korotovsky.io>
+ * @license WTFPL
+ */
+class PgAdvisoryMutex extends LockMutex
+{
+    /**
+     * @var \PDO $pdo The PDO.
+     */
+    private $pdo;
+
+    /**
+     * @var array The resource map <String, Integer>
+     */
+    private $map;
+
+    /**
+     * @var string
+     */
+    private $resourceType;
+
+    /**
+     * @var int
+     */
+    private $resourceId;
+
+    /**
+     * PgAdvisoryMutex constructor.
+     * @param \PDO    $pdo           The PDO connection
+     * @param array   $map           Resource map
+     * @param string  $resourceType  Name of the current lock key
+     * @param int     $resourceId    Resource id
+     */
+    public function __construct(\PDO $pdo, array $map, $resourceType, $resourceId = 0)
```

Comment:
There's no support for timeouts?

Reply:
Yes, there is no. It's non-blocking `SELECT` call of `pg_try_advisory_lock(key1 int, key2 int)` https://www.postgresql.org/docs/current/static/functions-admin.html#FUNCTIONS-ADVISORY-LOCKS-TABLE.

The lock becomes invalid once the database connection is broken/interrupted or terminated in a normal way. Another interesting thing: with the same parameters the call of `pg_try_advisory_lock()` would return always `t` (lock acquired) in the same database connection.